ts/features/bonus/cgn/components/CgnServiceCTA.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from "react";
+import { constNull } from "fp-ts/lib/function";
+import { Label } from "../../../../components/core/typography/Label";
+import ButtonDefaultOpacity from "../../../../components/ButtonDefaultOpacity";
+import I18n from "../../../../i18n";
+import { useIODispatch, useIOSelector } from "../../../../store/hooks";
+import { servicePreferenceSelector } from "../../../../store/reducers/entities/services/servicePreference";
+import { isStrictSome } from "../../../../utils/pot";
+import { isServicePreferenceResponseSuccess } from "../../../../types/services/ServicePreferenceResponse";
+import { ServiceId } from "../../../../../definitions/backend/ServiceId";
+import { loadAvailableBonuses } from "../../bonusVacanze/store/actions/bonusVacanze";
+import { cgnActivationStart } from "../store/actions/activation";
+
+type Props = {
+  serviceId: ServiceId;
+};
+const CgnServiceCTA = (props: Props) => {
+  const dispatch = useIODispatch();
+  const servicePreference = useIOSelector(servicePreferenceSelector);
+
+  const isServiceActive =
+    isStrictSome(servicePreference) &&
+    isServicePreferenceResponseSuccess(servicePreference.value) &&
+    servicePreference.value.value.inbox;
+
+  if (
+    !isStrictSome(servicePreference) ||
+    (isStrictSome(servicePreference) &&
+      servicePreference.value.id !== props.serviceId)
+  ) {
+    return <></>;
+  }
+
+  return isServiceActive ? (
```

Comment:
```suggestion
    const servicePreferenceValue = pot.getOrElse(servicePreference, undefined);

  if (
    !servicePreferenceValue ||
    servicePreferenceValue.id !== props.serviceId
  ) {
    return null;
  }
  const isServiceActive =
    servicePreferenceValue &&
    isServicePreferenceResponseSuccess(servicePreferenceValue) &&
    servicePreferenceValue.value.inbox;

  return isServiceActive ? (
```

this refactoring aims to:

- avoid 3 times computation of `isStrictSome(servicePreference)`
- compute `isServiceActive` when it is really needed

(I didn't test it)